cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,211 @@ See the Apache™ Twill®
+abstraction over data that is organized into files.
+
+Creating a PartitionedFileSet
+=============================
+
+To create and use a PartitionedFileSet in an application, you create it as part of the application
+configuration, similar to FileSets. However, the partitioning has to be given as an additional property::
+
+  public void configure() {
+
+    createDataset("results", PartitionedFileSet.class, PartitionedFileSetProperties.builder()
+      // Properties for partitioning
+      .setPartitioning(Partitioning.builder().addStringField("league").addIntField("season").build())
+      // Properties for file set
+      .setInputFormat(TextInputFormat.class)
+      .setOutputFormat(TextOutputFormat.class)
+      .setOutputProperty(TextOutputFormat.SEPERATOR, ",")
+      .build());
+
+This creates a new PartitionedFileSet named *results*. Similar to FileSets, it specifies ``TextInputFormat`` and
+``TextOutputFormat.``, and for the output format, we specify that the separator between fields is a comma.
+The difference to a FileSet is that this Dataset is partitioned by league and season. This means that every file
+added to this Dataset must have a partitioning key with a unique combination of league and season.
+
+Reading and Writing PartitionedFileSets
+=======================================
+
+You can interact with the files in a PartitionedFileSet directly, through the ``Location`` abstraction
+of the file system. This is similar to a FileSet, but instead of a relative path, you specify a
+partition key to obtain a Partition; you can then get a Location from that Partition.
+
+For example, to read the content of a partition::
+
+      PartitionKey key = PartitionKey.builder().addStringField(...)
+                                               .addIntField(...)
+                                               .build());
+      Partition partition = dataset.getPartition(key);
+      if (partition != null) {
+        try {
+          Location location = partition.getLocation();
+          InputStream inputStream = location.getInputStream();
+          ...
+        } catch (IOException e) {
+          ...
+        }
+      }
+
+Note that if the partition was written with MapReduce, the location is actually a directory
+that contains part files. In that case, list the files in the directory to find the part files::
+
+    for (Location file : location.list()) {
+      if (file.getName().startsWith("part")) {
+        InputStream inputStream = location.getInputStream();
+        ...
+      }
+    }
+
+Instead of reading a single partition, you can also specify a PartitionFilter to query the
+partitioned file set for all partitions whose keys match that filter. The PartitionFilter
+can specify either a exact value (en equality condition) or a range for the value of each
+field in the dataset's partitioning. For example, the following code reads all partitions
+for the NFL and the eighties' seasons::
```

Comment:
for the NFL and the eighties seasons:: \[I'm thinking that as you would say the "1982 season", should say the '80s seasons.]